src/main/java/com/dev/cinema/controller/ShoppingCartController.java

```diff
@@ -35,15 +37,26 @@ public ShoppingCartController(MovieSessionService movieSessionService,
     }
 
     @PostMapping("/movie-sessions")
-    public void addMovieSession(@RequestParam Long userId, @RequestParam Long movieSessionId) {
-        User user = userService.get(userId);
+    public void addMovieSession(@RequestParam Authentication authentication,
+                                @RequestParam Long movieSessionId) {
+        Object principal = authentication.getPrincipal();
+        UserDetails userDetails = null;
+        if (principal instanceof UserDetails) {
+            userDetails = (UserDetails) principal;
+        }
+        User user = userService.findByEmail(userDetails.getUsername()).get();
```

Comment:
same

Reply:
Seems to me that it was mentioned today. There is no need here in orElseThrow(), we can leave get() here.